content/300-guides/400-migrate-to-prisma/03-migrate-from-mongoose.mdx

```diff
@@ -0,0 +1,1056 @@
+
+    if (!category)
+      return res.status(404).json({ message: 'Category not found' })
+
+    const post = await Post.findByIdAndUpdate(
+      { _id: id },
+      {
+        categories: [{ _id: categoryId }],
+      },
+      { new: true }
+    )
+
+    if (!post) return res.status(404).json({ message: 'Post not found' })
+    return res.status(200).json(post)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+With Prisma, the handler is implemented as follows:
+
+```js file=src/controllers/post.js
+const addPostToCategory = async (req, res) => {
+  const { id, categoryId } = req.query
+
+  try {
+    const post = await prisma.post.update({
+      where: {
+        id,
+      },
+      data: {
+        categories: {
+          connect: {
+            id: categoryId,
+          },
+        },
+      },
+    })
+
+    if (!post) return res.status(404).json({ message: 'Post not found' })
+
+    return res.status(200).json(post)
+  } catch (error) {
+    console.log({ error })
+    return res.status(500).json(error)
+  }
+}
+```
+
+#### `/post/:id`
+
+The `/post/:id` handler is currently implemented as follows:
+
+```js file=src/controllers/post.js
+const publishDraft = async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const post = await Post.findByIdAndUpdate(
+      { id },
+      { published: true },
+      { new: true }
+    )
+    return res.status(200).json(post)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+With Prisma, the handler is implemented as follows:
+
+```js file=src/controllers/post.js
+const publishDraft = async (req, res) => {
+  const { id } = req.params
+
+  try {
+    const post = await prisma.post.update({
+      where: { id },
+      data: { published: true },
+    })
+    return res.status(200).json(post)
+  } catch (error) {
+    return res.status(500).json(error)
+  }
+}
+```
+
+## More
+
+### Embedded documents `_id` field
+
+By default, Mongoose assigns each document and embedded document an `_id` field. If you wish to disable this option for embedded documents, you can set the `_id` option to false.
+
+```js
+const ProfileSchema = new Schema(
+  {
+    bio: String,
+  },
+  {
+    _id: false,
+  }
+)
+```
+
+### Document version key
+
+Mongoose assigns each document a version when created. You can disable Mongoose from versioning your documents by setting the `versionKey` option of a model to false. It is [not recommended](http://aaronheckmann.blogspot.com/2012/06/mongoose-v3-part-1-versioning.html) to disable this unless you know what you're doing.
+
+```js
+const ProfileSchema = new Schema(
+  {
+    bio: String,
+  },
+  {
+    versionKey: false,
+  }
+)
+```
+
+When migrating to Prisma, mark the field as optional in your Prisma schema and add the `@ignore` attribute to exclude it from Prisma Client.
+
+### Collection name inference
+
+Mongoose infers the collection names by automatically lowercasing and pluralizing the model names. .
```

Comment:
Is this how the example went from "Category" to "categories"?

Reply:
I'm not sure I completely follow but in your `category.js` model file, it would be called `category` but when the collection is created, it's called `categories`